src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -194,14 +194,13 @@ Scalar<DataVector> spin_function(
     const Scalar<DataVector>& area_element,
     const tnsr::ii<DataVector, 3, Frame>& extrinsic_curvature) noexcept {
   auto sin_theta = make_with_value<Scalar<DataVector>>(area_element, 0.0);
```

Comment:
Not even close, you haven't changed anything at all.

```
template <typename Frame>
Scalar<DataVector> spin_function(
    const StrahlkorperTags::StrahlkorperTags_detail::Jacobian<Frame>& tangents,
    const YlmSpherepack& ylm,
    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector,
    const Scalar<DataVector>& area_element,
    const tnsr::ii<DataVector, 3, Frame>& extrinsic_curvature) noexcept {
  auto sin_theta = make_with_value<Scalar<DataVector>>(area_element, 0.0);

  auto extrinsic_curvature_theta_normal_sin_theta =
      make_with_value<Scalar<DataVector>>(area_element, 0.0);
  auto extrinsic_curvature_phi_normal =
      make_with_value<Scalar<DataVector>>(area_element, 0.0);
  for (size_t i = 0; i < 3; ++i) {
    get(sin_theta) = extrinsic_curvature.get(i, 0) * unit_normal_vector.get(0);
    for (size_t j = 1; j < 3; ++j) {
      get(sin_theta) +=
          extrinsic_curvature.get(i, j) * unit_normal_vector.get(j);
    }

    // Note: I must multiply by sin_theta because
    // I take the phi derivative of this term by using
    // the spherepack gradient, which includes a
    // sin_theta in the denominator of the phi derivative.
    // Will do this outside the i,j loops.
    get(extrinsic_curvature_theta_normal_sin_theta) +=
        get(sin_theta) * tangents.get(i, 0);

    // Note: I must multiply by sin_theta because tangents.get(i,1)
    // actually contains \partial_\phi / sin(theta), but I want just
    //\partial_\phi. Will do this outside the i,j loops.
    get(extrinsic_curvature_phi_normal) +=
        get(sin_theta) * tangents.get(i, 1);
  }
  get(sin_theta) = sin(ylm.theta_phi_points()[0]);
  get(extrinsic_curvature_theta_normal_sin_theta) *= get(sin_theta);
  get(extrinsic_curvature_phi_normal) *= get(sin_theta);

  get(sin_theta) =
      (ylm.gradient(extrinsic_curvature_phi_normal.get()).get(0) -
       ylm.gradient(extrinsic_curvature_theta_normal_sin_theta.get()).get(1)) /
      (sin_theta.get() * area_element.get());

  return sin_theta;
}
```

Reply:
I'm all for reducing allocations by reusing the same DataVector, but if `sin_theta` isn't always holding sin(theta), then I'd favor renaming it to something more honest like `temp` or `buffer`. Otherwise this will be misleading to the reader.

Reply:
@nilsdeppe your comment "not even close" doesn't help me understand the problem, and it creates a needlessly hostile working environment. 

When you say "use sin_theta allocation," can you (or someone else) please clarify what you mean by this? I (apparently incorrectly) took it to mean initialize with sin_theta instead of area_element (I don't understand why that would matter). 

As for the "compute sin_theta line above here", that sounded like move the definition of sin(theta) to right before it is used, so I did. Is that not what you meant? If not, again, can you please clarify?

Reply:
@geoffrey4444 it means to allocate a single DataVector that is used for multiple purposes in order to avoid reallocating memory which an expensive operation.  I agree with @fmahebert that in such a case it should probably be named something like temp, where if it helps for clarity you can use named  references to temp e.g. `DataVector& sin_theta = temp`.  You are correct that @nilsdeppe was asking you to move the call to fill sin_theta to right before it is used.